### PR TITLE
Refactor img helper

### PR DESCRIPTION
**Pull request by NataliaLKB**
## What does this change?

There are some places where you don't want to use the whole scala img helper, you just want to use `<picture>` element. This change makes that much easier. 

I will mention that @OliverJAsh 's PR https://github.com/guardian/frontend/pull/12706 can also use this instead of the current `` being used.
## What is the value of this and can you measure success?

Makes the code a bit more dry, and hopefully easier to read (renaming to `imageFigure` and `image` instead of just `img`)

### CODE REVIEW

**Comment by NataliaLKB** [comment]
As a side benefit, this PR also reserves the space in Galleries for all images before they load, greatly reducing JANK.

File: applications/app/views/fragments/galleryHeader.scala.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
 <header class="content__head is-fixed-height tonal__head tonal__head--@toneClass(gallery.item)">
     @* Hidden because we visually show this data in the head, but needed
     here for SEO. *@
-    <h1 class="is-hidden" itemprop="headline" class="is-hidden">@Html(gallery.item.trail.headline)</h1>
```
Comment (OliverJAsh) [inline applications/app/views/fragments/galleryHeader.scala.html]: Oops, thanks!

File: common/app/views/fragments/image.scala.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    @picture.largestImage.map{ largestImage =>
+        <div
+        class="u-responsive-ratio@if(isFeature && isShowcase) { u-responsive-ratio--letterbox}"
+            @if(isFeature && isShowcase) {} else {style="padding-bottom: @{"%.2f".format(100 * largestImage.height.toDouble / largestImage.width)}%"}
+        >
+    }
+
+    <picture>
+        @* IE 9 needs this workaround as per https://scottjehl.github.io/picturefill/ *@
+            <!--[if IE 9]><video style="display: none;"><![endif]-->
```
Comment (OliverJAsh) [inline common/app/views/fragments/image.scala.html]: Indentation is a bit off here and the IE comment below.

File: common/app/views/main.scala.html [hunk 3]
```diff
@@ -31,8 +31,7 @@
         @defining(
             page.item.content.trail.trailPicture.head
         ) { picture =>
-            <img src="http:@Html(ImgSrc.findNearestSrc(picture, Profile(width = Some(2200))).getOrElse(""))"
-            class="immersive-main-media__img">
+            @fragments.image(picture, imageAltText = "")
```
Comment (OliverJAsh) [inline common/app/views/main.scala.html]: We could use the caption as alt text?

File: common/app/views/fragments/image.scala.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
+@import layout.ContentWidths.MainMedia
+@import views.support.ImgSrc
+
+@(
+    picture: model.ImageMedia,
+    isShowcase: Boolean = false,
+    isFeature: Boolean = false,
+    widths: layout.WidthsByBreakpoint =  MainMedia.inline,
```
Comment (OliverJAsh) [inline common/app/views/fragments/image.scala.html]: Galleries main media will currently be using the default `widths` specified here. Do we need to specify our own `widths`?

File: applications/app/views/fragments/galleryHeader.scala.html (hunk 1 above)
Comment (NataliaLKB) [inline applications/app/views/fragments/galleryHeader.scala.html]: 😉

File: common/app/views/main.scala.html (hunk 3 above)
Comment (NataliaLKB) [inline common/app/views/main.scala.html]: Would this mean the caption would be read twice?

**Comment by OliverJAsh** [comment]
I think there are two(?) other instances of the picture el in our source code. Could we de-dupe those too?

**Comment by NataliaLKB** [comment]
There is one other instance of `<picture>` used by facia cards. The parameters are quite a bit different, which is probably why it wasn't using the img helper to begin with. 

It would add much more complexity to the now simple image helper. I am not sure if its worth it.

File: common/app/views/fragments/image.scala.html (hunk 4 above)
Comment (NataliaLKB) [inline common/app/views/fragments/image.scala.html]: Thanks, this has been addressed 😄

File: common/app/views/main.scala.html (hunk 3 above)
Comment (OliverJAsh) [inline common/app/views/main.scala.html]: I think we also need a `widths` value here, we'll probably have to create a new one?

Comment (NataliaLKB) [inline common/app/views/main.scala.html]: Yes agreed. I will add that to this PR.

File: applications/app/views/fragments/newGalleryBody.scala.html [hunk 5]
```diff
@@ -74,30 +74,7 @@ <h2 class="fc-container__header__title">
...
-                            sizes="@breakpointWidth.width"
-                            srcset="@ImgSrc.srcsetForBreakpoint(breakpointWidth, widths.breakpoints, maybePath = None, maybeImageMedia = Some(imageElement.images))" />
-                        }
-                    }
-
-                    <!--[if IE 9]></video><![endif]-->
-
-                    <img class="gallery__img gallery__img@if(image.width >= image.height) {--landscape} else {--portrait}"
-                    itemprop="contentUrl"
-                        style="max-width: calc(@{image.width.toFloat/image.height} * 96vh)"
```
Comment (OliverJAsh) [inline applications/app/views/fragments/newGalleryBody.scala.html]: As discussed, let's make sure the max-width still applies with the new img

File: common/app/views/main.scala.html (hunk 3 above)
Comment (OliverJAsh) [inline common/app/views/main.scala.html]: Related: I was thinking perhaps we should rename `ImmersiveMedia.immersive` to `ImmersiveMedia.inline`?

Comment (OliverJAsh) [inline common/app/views/main.scala.html]: Actually ignore that!

File: applications/app/views/fragments/newGalleryBody.scala.html (hunk 5 above)
Comment (NataliaLKB) [inline applications/app/views/fragments/newGalleryBody.scala.html]: This has been fixed, please take a look when you have time @OliverJAsh

**Comment by OliverJAsh** [comment]
👍

**Comment by OliverJAsh** [comment]
Awesome work 😄

**Comment by NataliaLKB** [comment]
😊

**Pull request merged**